core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -276,7 +276,15 @@ private boolean usePendingAcquisition(SessionProtocol desiredProtocol, PoolKey k
             if (cause == null) {
                 final SessionProtocol actualProtocol = pch.protocol();
                 if (actualProtocol.isMultiplex()) {
-                    promise.complete(pch);
+                    final HttpSession session = HttpSession.get(pch.get());
+                    logger.info("session.unfinishedResponses(): {}, session.maxUnfinishedResponses(): {}",
+                                session.unfinishedResponses(), session.maxUnfinishedResponses());
+                    // Need to subtract 1 since the current pending request may also be unfinished
+                    if (session.unfinishedResponses() >= session.maxUnfinishedResponses() - 1) {
+                        acquireLater(actualProtocol, key, timingsBuilder).thenAccept(promise::complete);
```

Comment:
Shouldn't we call `acquireLater()` after `pendingAcquisition` is cleared? `acquireLater()` (and inherently `usePendingAcquisition()`) would use the current `pendingAcquisition` again and again, making a busy loop.

Reply:
I see~ Of course I'll explore this idea, but just to share my original thought.
I was thinking `pendingAcquisition` would always be cleared since `notifyConnect` clears it before completing the promise.

Reply:
Ah, OK. `pendingAcquisition` is removed in `notifyConnect()` before `pendingAcquisition` is completed.